src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupLoggingHelper.cs

```diff
@@ -271,6 +272,11 @@ private static void AppendStringFromParameterValue(ReuseableStringBuilder sb, ob
             bool logItemMetadata,
             DateTime timestamp)
         {
+            // Only run this method if we use AppDomains and not in the default AppDomain
+#if FEATURE_APPDOMAIN
```

Comment:
This only checks that we might be in an app domain rathe than that we necessarily are, right? Is there any way to tell that we are? Also, maybe remove the part of the comment about default app domain? I was confused until I looked at what CreateItemsSnapshot actually does.